### PR TITLE
Add block support to form

### PR DESCRIPTION
**Pull request by danrot**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | ---
| Related issues/PRs | ---
| License | MIT
| Documentation PR | ---

#### What's in this PR?

This PR will wrap the `BlockCollection` and build a `FieldType` for the Form based on that.

#### Why?

Because we need support for repeatable blocks in forms.

#### To Do

- [ ] Create a documentation PR

### CODE REVIEW

**Comment by danrot** [comment]
@chirimoya This is finished except for the fact that it is still called `FormBlockCollection`. And I'll wait with writing the README until we have decided on the final name :wink:

**Comment by chirimoya** [comment]
please add an issue for:
- display error of the `media_selection` field type
- block preview is missing

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FormRenderer.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+type Props = {
+    data: Object,
+    index: number,
+    locale: ?IObservableValue<string>,
+    onChange: (index: number, name: string, value: *) => void,
+    schema: Schema,
+};
+
+export default class FormRenderer extends React.Component<Props> {
```
Comment (chirimoya) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FormRenderer.js]: `FieldRenderer`

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/Form.js [hunk 2]
```diff
@@ -13,41 +14,47 @@ type Props = {
...
-                onSubmit={this.handleSubmit}
-                onChange={this.handleChange}
-                schema={store.schema}
-                data={store.data}
-                locale={store.locale}
-            />;
+            : (
+                <form onSubmit={this.handleSubmit}>
+                    <Renderer
+                        onSubmit={this.handleSubmit}
```
Comment (chirimoya) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/Form.js]: remove `onSubmit` property

**Comment by danrot** [comment]
Created issues #3770 and #3771. Already contacted Tim regarding the MediaSelection in the block.

**Pull request merged**